pkg/cmd/extension/manager_test.go

```diff
@@ -613,6 +613,10 @@ func TestManager_Install_git(t *testing.T) {
 	ios, _, stdout, stderr := iostreams.Test()
 	m := newTestManager(tempDir, &client, ios)
 
+	reg.Register(
+		httpmock.REST("GET", "owner/gh-some-ext"),
```

Comment:
As evident by these new stubs in every existing test, `repoExists` is now called in places where it was previously not needed, adding an overhead of an extra API request without adding any value— unless repo in fact does not exist, in which case it raises a better error message.

Would it be an option to call `repoExists` only in cases when fetching a release or fetching repo file contents returned a 404, to clarify what exactly was missing?

Reply:
I directed us down this current path but perhaps this is a better suggestion.

Reply:
I can take a look later to see where I can push this further down the stack to only the places it is needed